src/langchain_google_alloydb_pg/utils/pgvector_migrator.py

```diff
@@ -0,0 +1,320 @@
+        collection_data_len = result_map.fetchone()
+    if collection_data_len is None:
+        warnings.warn(f"Collection, {collection_name} contains no elements.")
+        return
+
+    # Extract data from the collection and batch insert into the new table
+    data_batches = _aextract_pgvector_collection(
+        engine, collection_name, batch_size=insert_batch_size
+    )
+
+    pending: set[Any] = set()
+    max_concurrency = 5
+    async for batch_data in data_batches:
+        if len(pending) == max_concurrency:
+            _, pending = await asyncio.wait(
+                pending, return_when=asyncio.FIRST_COMPLETED
+            )
+        else:
+            pending.add(
+                asyncio.ensure_future(
+                    vector_store.aadd_embeddings(
+                        texts=[data.document for data in batch_data],
+                        embeddings=[data.embedding for data in batch_data],
+                        metadatas=[data.cmetadata for data in batch_data],
+                        ids=[data.id for data in batch_data],
+                    )
+                )
+            )
+    await asyncio.wait(pending)
+
+    # Validate data migration
+    query = f"SELECT COUNT(*) FROM {destination_table}"
+    async with engine._pool.connect() as conn:
+        result = await conn.execute(text(query))
+        result_map = result.mappings()
+        table_size = result_map.fetchone()
+    if not table_size:
+        raise ValueError(f"Table: {destination_table} does not exist.")
+
+    if collection_data_len["count"] != table_size["count"]:
+        raise ValueError(
+            "All data not yet migrated.\n"
+            f"Original row count: {collection_data_len['count']}\n"
+            f"Collection table, {destination_table} row count: {table_size['count']}"
+        )
+    elif delete_pg_collection:
+        # Delete PGVector data
+        query = f"DELETE FROM {EMBEDDINGS_TABLE} WHERE collection_id='{uuid}'"
+        async with engine._pool.connect() as conn:
+            await conn.execute(text(query))
+            await conn.commit()
+
+        query = f"DELETE FROM {COLLECTIONS_TABLE} WHERE name='{collection_name}'"
+        async with engine._pool.connect() as conn:
+            await conn.execute(text(query))
+            await conn.commit()
+        print(f"Successfully deleted PGVector collection, {collection_name}")
+
+
+async def _alist_pgvector_collection_names(
+    engine: AlloyDBEngine,
+) -> List[str]:
+    """Lists all collection names present in PGVector table."""
+    try:
+        query = f"SELECT name from {COLLECTIONS_TABLE}"
+        async with engine._pool.connect() as conn:
+            result = await conn.execute(text(query))
+            result_map = result.mappings()
+            all_rows = result_map.fetchall()
+        return [row["name"] for row in all_rows]
+    except ProgrammingError as e:
+        raise ValueError("Please provide the correct collection table name: " + str(e))
```

Comment:
nit: Considering using "from" to transmit "cause" in case ProgrammingError might have other causes 
```suggestion
        raise ValueError("Please provide the correct collection table name: " + str(e)) from e
```